### PR TITLE
Adding ylearn

### PR DESCRIPTION
**Pull request by lixfz**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by lixfz** [comment]
@conda-forge/help-python Ready for review! Thanks

**Comment by lixfz** [comment]
@conda-forge/help-python and @conda-forge/help-python-c  

Ready for review! Thanks

**Comment by lixfz** [comment]
@conda-forge/staged-recipes  ready for review!

File: recipes/ylearn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/DataCanvasIO/{{ name }}/archive/refs/tags/{{ version }}.tar.gz
+  sha256: c54fa3e18ad6dce9e7ac0997db809572e7ebe2cf22be502aabcc84c3282786f6
+
+build:
+  number: 0
```
Comment (ocefpaf) [inline recipes/ylearn/meta.yaml:13]: You can remove the build scripts and call. If the tarball is OK you don't need to inplace build step.

```suggestion
  script: {{ PYTHON }} -m pip install . -vv
  number: 0
```

Comment (lixfz) [reply recipes/ylearn/meta.yaml:14]: Thank you for your suggestion.
I remove the build scripts and switch tarball source  to pypi.

File: recipes/ylearn/meta.yaml [hunk 2]
```diff
@@ -6,17 +6,17 @@ package:
...
 
 build:
+  script: {{ PYTHON }} -m pip install . -vv
   number: 0
 
 requirements:
   build:
     - {{ compiler('cxx') }}
   host:
-    - cython
```
Comment (ocefpaf) [inline recipes/ylearn/meta.yaml]: I did not check upstream's setup.py but re-cythonizing should be an option.
I'll leave that to you in the feedstock. It is preferred b/c it makes migrations to new python versions easier.

Comment (lixfz) [reply recipes/ylearn/meta.yaml]: Thanks! 
To support re-cythonizing, the setup settings need to be updated.  I'll  fix it  at next release.

**Comment by lixfz** [comment]
@ocefpaf Thank you for your review and help.

**Pull request merged**